src/util/strencodings.h

```diff
@@ -69,7 +71,25 @@ std::string EncodeBase32(const std::string& str, bool pad = true);
 
 void SplitHostPort(std::string in, int& portOut, std::string& hostOut);
 int64_t atoi64(const std::string& str);
-int atoi(const std::string& str);
+
+template <typename T>
+T ToIntegral(const std::string& str)
+{
+    T result;
+    // Emulate atoi(...) handling of white space and leading +/-.
+    std::string s = TrimString(str);
+    if (!s.empty() && s[0] == '+') {
+        if (s.length() >= 2 && s[1] == '-') {
+            return 0;
+        }
+        s = s.substr(1);
+    }
+    auto [_, error_condition] = std::from_chars(s.data(), s.data() + s.size(), result);
+    if (error_condition != std::errc{}) {
+        return 0;
```

Comment:
Please, do not just return 0 in case of a parse error but use explicit error feedback. Most C++17ish would be to return an `Option`.

Reply:
I think you need to check the returned pointer against `last` as well, otherwise it ignores any remainder.

Reply:
I agree on both for a sane safe interface, but since this function (`LocaleIndependentAtoi`) is meant as a quirk-by-quirk compatible to the not-entirely-sane `atoi(…)` I'm afraid we need to return a `T` (instead of `std::optional<T>`) and ignore any remainder, no? :)